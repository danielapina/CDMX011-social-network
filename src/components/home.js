// eslint-disable-next-line import/no-cycle
import { onNavigate } from '../main.js';

export const home = () => {
  const html = `
  <header>
  <a href="/"><img src="img/logo-desktop.png" alt="logotipo" id="logoGF" /></a>
  <span class="menu-icon" id="users-icon"><img id="img-users" src="img/users.png" alt=""></span>

  <div class="div-navegation">
    <ul id="navegacion">
      <li><a><button class="blue btn-routing" id="btn-register" value='/register'>¡Regístrate!</button></a></li>
      <li><a><button class="blue btn-routing" id="btn-login" value='/login'>Iniciar Sesión</button></a></li>
    </ul>
  </div>
</header>

    <div id="home-page">
      <div class="banner">
        <p class="banner-welcome">
          Cuidar de la naturaleza es cuidar de nosotros.
        </p>
        <button class="btns-login btn-routing" id='btn-register-movil' value='/register'>¡Regístrate!</button>
      </div>
      <div class="process">
        <h2 id="title-process">¿Cómo Funciona?</h2>
        <div class="process-img">
          <img id="process-movil" src="img/process-movil.png" alt="register" />
          <img id="process-desktop" src="img/process-desktop.png" alt="register" />
        </div>
      </div>
    </div>
   
   `;

  const divHome = document.createElement('div');
  divHome.innerHTML = html;

  const btnRegister = divHome.querySelector('#btn-register');
  btnRegister.addEventListener('click', () => onNavigate('/register'));

  const btnLogin = divHome.querySelector('#btn-login');
  btnLogin.addEventListener('click', () => onNavigate('/login'));

  const btnRegisterMovil = divHome.querySelector('#btn-register-movil'); // Esta apunta a divHome;
  btnRegisterMovil.addEventListener('click', () => onNavigate('/register'));

  // -------------------------------icon burguer
  divHome.querySelector('#users-icon').addEventListener('click', () => {
    const mostrar = divHome.querySelector('#navegacion').className;
    if (mostrar === '') {
      divHome.querySelector('#navegacion').setAttribute('class', 'show');
    } else {
      divHome.querySelector('#navegacion').removeAttribute('class', 'show');
    }
  });

  return divHome;
};
