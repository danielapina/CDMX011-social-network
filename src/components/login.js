// eslint-disable-next-line import/no-cycle
import { onNavigate } from '../main.js';

export const login = () => {
  const html = `
  <header>
  <a href="/"><img src="img/logo-desktop.png" alt="logotipo" id="logoGF" /></a>
  <span class="menu-icon" id="users-icon"><img id="img-users" src="img/users.png" alt=""></span>
  <div class="div-navegation">
    <ul id="navegacion">
      <li><a><button class="blue btn-routing" id="btn-register" value='/register'>¡Regístrate!</button></a></li>
      <li><a><button class="blue btn-routing" id="btn-login" value='/login'>Login</button></a></li>
    </ul>
  </div>
</header>
  <div id="login-page">
     <form  id="login-form" class ="form-inicial">
       <h2 class="titles" id="title-form">¡Hola Green Friend!</h2>
       <img id="heart-movil" src="img/heart.png" alt="heart" />
       <label for="email">Correo electronico</label>
       <input type="email" id="login-email" placeholder='alguien@ejemplo.com'/>
       <label for="password">Contraseña</label>
       <div class = "show-passwords">
       <input type="password" id="login-password" placeholder="******"/>
       <span class = "eyes-login">
       <i id="show3" class="fas fa-eye" style="color: #0e6359;"></i>
       <i id="hide3" class="fas fa-eye-slash" style="color: #0e6359;"></i>
       </span>
       </div>
       <br>
       <button id="form-button" class="submit-btn">Enviar</button>
       <button id="btn-google"class="submit-btn google"><img src="img/google.png" alt="google" id="google-icon">Login</button>
     </form>
     <div class="img-register-desktop">
       <img id="heart" src="img/heart.png" alt="heart" />
     </div>
   </div>
  `;
  const divLogin = document.createElement('div');
  divLogin.innerHTML = html;

  const btnRegister = divLogin.querySelector('#btn-register');
  btnRegister.addEventListener('click', () => onNavigate('/register'));
  const btnLogin = divLogin.querySelector('#btn-login');
  btnLogin.addEventListener('click', () => onNavigate('/login'));

  // -------------------------------icon burguer
  divLogin.querySelector('#users-icon').addEventListener('click', () => {
    const mostrar = divLogin.querySelector('#navegacion').className;
    if (mostrar === '') {
      divLogin.querySelector('#navegacion').setAttribute('class', 'show');
    } else {
      divLogin.querySelector('#navegacion').removeAttribute('class', 'show');
    }
  });

  const loginForm = divLogin.querySelector('#login-form');
  loginForm.addEventListener('submit', (e) => {
    e.preventDefault();
    const email = divLogin.querySelector('#login-email').value;
    const password = divLogin.querySelector('#login-password').value;
    console.log(email, password);

    const auth = firebase.auth();
    auth
      .signInWithEmailAndPassword(email, password)
      .then((userCredential) => {
        onNavigate('/profile');
        console.log(userCredential);
      })
      .catch((error) => {
        const errorMessage = error.message;
        alert(errorMessage);
      });
  });
  // GOOGLE
  const auth = firebase.auth();
  const btnGoogle = divLogin.querySelector('#btn-google');
  btnGoogle.addEventListener('click', (e) => {
    e.preventDefault();
    const provider = new firebase.auth.GoogleAuthProvider();
    auth.signInWithPopup(provider)
      .then((result) => {
        console.log('Registro con google', result);
        onNavigate('/profile');
      })
      .catch((err) => { alert(err); });
  });
  // -------------------------------icon mostrar password de login
  divLogin.querySelector('.eyes-login').addEventListener('click', (e) => {
    e.preventDefault();
    const passwordInput = divLogin.querySelector('#login-password');
    const hide = divLogin.querySelector('#hide3');
    const show = divLogin.querySelector('#show3');
    if (passwordInput.type === 'password') {
      passwordInput.type = 'text';
      hide.style.display = 'block';
      show.style.display = 'none';
    } else {
      passwordInput.type = 'password';
      hide.style.display = 'none';
      show.style.display = 'block';
    }
  });

  return divLogin;
};
