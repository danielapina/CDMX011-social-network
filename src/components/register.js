import { signUp } from '../lib/firebaseClient.js';
// eslint-disable-next-line import/no-cycle
import { onNavigate } from '../main.js';

export const register = () => {
  const html = `
  <header>
  <a href="/"><img src="img/logo-desktop.png" alt="logotipo" id="logoGF" /></a>
  <span class="menu-icon" id="users-icon"><img id="img-users" src="img/users.png" alt=""></span>
  <div class="div-navegation">
    <ul id="navegacion">
      <li><a><button class="blue btn-routing" id="btn-login" value='/login'>Iniciar Sesión</button></a></li>
    </ul>
  </div>
 </header>
  <div id="register-page">
     <form id="register-form" class ="form-inicial" >
      <img class="leaf-img" src="img/leafs-desktop.png" alt="leafs" />
       <h2 class="titles" id="title-form">Registro</h2>
       <label for="email">Correo electrónico</label>
       <input type="email" id="user-email" placeholder='alguien@ejemplo.com'/>
       <label for="password">Contraseña</label>
       <div class = "show-passwords">
       <input type="password" id="user-password" placeholder='Mínimo 6 carácteres' />
       <span class = "eyes">
       <i id="show1" class="fas fa-eye" style="color: #0e6359;"></i>
       <i id="hide1" class="fas fa-eye-slash" style="color: #0e6359;"></i>
       </span>
       </div>
       <label for="password"> Confirma Contraseña</label>
       <div class = "show-passwords">
       <input type="password" id="confirm-password" placeholder='Mínimo 6 carácteres' />
       <span class = "eyes-confirm">
       <i id="show2" class="fas fa-eye" style="color: #0e6359;"></i>
       <i id="hide2" class="fas fa-eye-slash" style="color: #0e6359;"></i>
       </span>
       </div>
       <br>
       <p id="error-message-register"></p>
       <button id="form-button-register"class="submit-btn">Enviar</button>
       <button id="btn-google"class="submit-btn google"><img src="img/google.png" alt="google" id="google-icon">Regístrate</button>
     </form>
     <div class="img-register-desktop">
       <img id="madre-tierra" src="img/madreTierra.png" alt="MadreTierra" />
     </div>
   </div>
`;
  const divRegister = document.createElement('div');
  divRegister.innerHTML = html;
  const btnLogin = divRegister.querySelector('#btn-login');
  btnLogin.addEventListener('click', () => onNavigate('/login'));

  // -------------------------------icon burguer
  divRegister.querySelector('#users-icon').addEventListener('click', () => {
    const mostrar = divRegister.querySelector('#navegacion').className;
    if (mostrar === '') {
      divRegister.querySelector('#navegacion').setAttribute('class', 'show');
    } else {
      divRegister.querySelector('#navegacion').removeAttribute('class', 'show');
    }
  });

  const btnForm = divRegister.querySelector('#register-form');
  btnForm.addEventListener('submit', (e) => {
    e.preventDefault();
    const email = divRegister.querySelector('#user-email').value;
    const password = divRegister.querySelector('#user-password').value;
    const confirmPassword = divRegister.querySelector('#confirm-password').value;
    if (password === confirmPassword) {
      signUp(email, password)
        .then(() => {
          onNavigate('/wall');
        })
        .catch((error) => {
          const errorMessage = error.message;
          divRegister.querySelector('#error-message-register').innerHTML = errorMessage;
        });
    } else {
      divRegister.querySelector('#error-message-register').innerHTML = 'Las Contraseñas no coinciden, vuelve a intentar.';
    }
  });

  // GOOGLE
  const auth = firebase.auth();
  const btnGoogle = divRegister.querySelector('#btn-google');
  btnGoogle.addEventListener('click', (e) => {
    e.preventDefault();
    const provider = new firebase.auth.GoogleAuthProvider();
    auth.signInWithPopup(provider)
      .then(() => {
        onNavigate('/wall');
      })
      .catch((err) => { divRegister.querySelector('#error-message-register').innerHTML = err; });
  });

  // -------------------------------icon mostrar password
  divRegister.querySelector('.eyes').addEventListener('click', (e) => {
    e.preventDefault();
    const passwordInput = divRegister.querySelector('#user-password');
    const hide = divRegister.querySelector('#hide1');
    const show = divRegister.querySelector('#show1');
    if (passwordInput.type === 'password') {
      passwordInput.type = 'text';
      hide.style.display = 'block';
      show.style.display = 'none';
    } else {
      passwordInput.type = 'password';
      hide.style.display = 'none';
      show.style.display = 'block';
    }
  });

  // -------------------------------icon mostrar confirm-password
  divRegister.querySelector('.eyes-confirm').addEventListener('click', (e) => {
    e.preventDefault();
    const passwordInput = divRegister.querySelector('#confirm-password');
    const hide = divRegister.querySelector('#hide2');
    const show = divRegister.querySelector('#show2');
    if (passwordInput.type === 'password') {
      passwordInput.type = 'text';
      hide.style.display = 'block';
      show.style.display = 'none';
    } else {
      passwordInput.type = 'password';
      hide.style.display = 'none';
      show.style.display = 'block';
    }
  });

  return divRegister;
};
