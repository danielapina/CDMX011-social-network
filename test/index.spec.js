/**
 * @jest-environment jsdom
 */
import './global/firebase.js';
// import './global/firebaseRegister.js';
import { home } from '../src/components/home.js';
import { login } from '../src/components/login.js';
import { register } from '../src/components/register.js';

describe('Coleccion de test sobre HOME', () => { // ----HOME---------------------------------------------------
  const homeRender = home();
  document.body.innerHTML = '<div id="root"></div>';
  test('should render', () => {
    const rootDiv = document.getElementById('root');
    rootDiv.appendChild(homeRender);
    expect(rootDiv.innerHTML).toMatchSnapshot();
  });
  test('Button register', () => {
    const btnRegister = homeRender.querySelector('#btn-register');
    btnRegister.click();
    expect(btnRegister.outerHTML).toBe('<button class="blue btn-routing" id="btn-register" value="/register">¡Regístrate!</button>');
  });
  test('Icon Burguer', () => {
    const btnIcon = homeRender.querySelector('#users-icon');
    btnIcon.click();
    expect(btnIcon.outerHTML).toBe('<span class="menu-icon" id="users-icon"><img id="img-users" src="img/users.png" alt=""></span>');
  });
  test('Remove show de Icon Burguer', () => {
    const btnIcon = homeRender.querySelector('#users-icon');
    btnIcon.click();
    btnIcon.click();
    expect(btnIcon.outerHTML).toBe('<span class="menu-icon" id="users-icon"><img id="img-users" src="img/users.png" alt=""></span>');
  });
});

describe('Coleccion de test sobre LOGIN', () => { // -------LOGIN----------------------------------------------
  document.body.innerHTML = '<div id="root"></div>';
  const component = login();
  test('should render', () => {
    const rootDiv = document.getElementById('root');
    rootDiv.appendChild(component);
    expect(rootDiv.innerHTML).toMatchSnapshot();
  });
  test('should logIn user when the submit button is clicked', () => {
    const mockLogin = jest.fn();
    mockLogin.mockImplementation(() => Promise.resolve());

    firebase.auth = jest.fn().mockImplementation(() => ({
      signInWithEmailAndPassword: mockLogin,
    }));
    const rootDiv = document.getElementById('root');
    rootDiv.appendChild(component);

    const email = 'test@laboratoria.la';
    const password = '123456';
    document.getElementById('login-email').value = email;
    document.getElementById('login-password').value = password;

    document.getElementById('form-button-login').click();

    expect(mockLogin).toHaveBeenCalledWith(email, password);
  });
  test('Icon Burguer', () => {
    const btnIcon = component.querySelector('#users-icon');
    btnIcon.click();
    expect(btnIcon.outerHTML).toBe('<span class="menu-icon" id="users-icon"><img id="img-users" src="img/users.png" alt=""></span>');
  });
  test('Remove show de Icon Burguer LOGIN', () => {
    const btnIcon = component.querySelector('#users-icon');
    btnIcon.click();
    expect(btnIcon.outerHTML).toBe('<span class="menu-icon" id="users-icon"><img id="img-users" src="img/users.png" alt=""></span>');
  });
  test('Eyes opened show password', () => {
    const btnEyes = component.querySelector('.eyes-login');
    btnEyes.click();
    expect(btnEyes.outerHTML).toBe(
      `<span class="eyes-login">
       <i id="show3" class="fas fa-eye" style="color: rgb(14, 99, 89); display: none;"></i>
       <i id="hide3" class="fas fa-eye-slash" style="color: rgb(14, 99, 89); display: block;"></i>
       </span>`,
    );
  });
  test('Eyes NO SHOW', () => {
    const btnEyes = component.querySelector('.eyes-login');
    btnEyes.click();
    expect(btnEyes.outerHTML).toBe(
      `<span class="eyes-login">
       <i id="show3" class="fas fa-eye" style="color: rgb(14, 99, 89); display: block;"></i>
       <i id="hide3" class="fas fa-eye-slash" style="color: rgb(14, 99, 89); display: none;"></i>
       </span>`,
    );
  });
});

describe('Coleccion de test soble REGISTER', () => { // -------REGISTER-------------------------------------------
  document.body.innerHTML = '<div id="root"></div>';
  const component = register();
  test('should render', () => {
    const rootDiv = document.getElementById('root');
    rootDiv.appendChild(component);
    expect(rootDiv.innerHTML).toMatchSnapshot();
  });
  test('should logIn user when the submit button is clicked', () => {
    const mockRegister = jest.fn();
    mockRegister.mockImplementation(() => Promise.resolve());

    firebase.auth = jest.fn().mockImplementation(() => ({
      createUserWithEmailAndPassword: mockRegister,
    }));
    const rootDiv = document.getElementById('root');
    rootDiv.appendChild(component);

    const email = 'test@laboratoria.la';
    const password = '123456';
    const confirmPassword = '123456';
    document.getElementById('user-email').value = email;
    document.getElementById('user-password').value = password;
    document.getElementById('confirm-password').value = confirmPassword;

    document.getElementById('form-button-register').click(); // Aqui llaama al form de Login
    expect(mockRegister).toHaveBeenCalledWith(email, password);
  });
  test('Icon Burguer', () => {
    const btnIcon = component.querySelector('#users-icon');
    btnIcon.click();
    expect(btnIcon.outerHTML).toBe('<span class="menu-icon" id="users-icon"><img id="img-users" src="img/users.png" alt=""></span>');
  });
  test('Remove show de Icon Burguer REGISTER', () => {
    const btnIcon = component.querySelector('#users-icon');
    btnIcon.click();
    console.log(btnIcon.outerHTML);
    expect(btnIcon.outerHTML).toBe('<span class="menu-icon" id="users-icon"><img id="img-users" src="img/users.png" alt=""></span>');
  });
  test('Eyes opened show password 1st', () => {
    const btnEyes = component.querySelector('.eyes');
    btnEyes.click();
    expect(btnEyes.outerHTML).toBe(
      `<span class="eyes">
       <i id="show1" class="fas fa-eye" style="color: rgb(14, 99, 89); display: none;"></i>
       <i id="hide1" class="fas fa-eye-slash" style="color: rgb(14, 99, 89); display: block;"></i>
       </span>`,
    );
  });
  test('Eyes NO SHOW 1st', () => {
    const btnEyes = component.querySelector('.eyes');
    btnEyes.click();
    expect(btnEyes.outerHTML).toBe(
      `<span class="eyes">
       <i id="show1" class="fas fa-eye" style="color: rgb(14, 99, 89); display: block;"></i>
       <i id="hide1" class="fas fa-eye-slash" style="color: rgb(14, 99, 89); display: none;"></i>
       </span>`,
    );
  });
  test('Eyes opened show password confirm', () => {
    const btnEyes = component.querySelector('.eyes-confirm');
    btnEyes.click();
    expect(btnEyes.outerHTML).toBe(
      `<span class="eyes-confirm">
       <i id="show2" class="fas fa-eye" style="color: rgb(14, 99, 89); display: none;"></i>
       <i id="hide2" class="fas fa-eye-slash" style="color: rgb(14, 99, 89); display: block;"></i>
       </span>`,
    );
  });
  test('Eyes NO SHOW confirm', () => {
    const btnEyes = component.querySelector('.eyes-confirm');
    btnEyes.click();
    expect(btnEyes.outerHTML).toBe(
      `<span class="eyes-confirm">
       <i id="show2" class="fas fa-eye" style="color: rgb(14, 99, 89); display: block;"></i>
       <i id="hide2" class="fas fa-eye-slash" style="color: rgb(14, 99, 89); display: none;"></i>
       </span>`,
    );
  });
});
