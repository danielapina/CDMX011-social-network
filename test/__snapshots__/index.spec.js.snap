// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coleccion de test soble REGISTER should render 1`] = `
"<div>
  <header>
  <a href=\\"/\\"><img src=\\"img/logo-desktop.png\\" alt=\\"logotipo\\" id=\\"logoGF\\"></a>
  <span class=\\"menu-icon\\" id=\\"users-icon\\"><img id=\\"img-users\\" src=\\"img/users.png\\" alt=\\"\\"></span>

  <div class=\\"div-navegation\\">
    <ul id=\\"navegacion\\" class=\\"show\\">
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-register\\" value=\\"/register\\">¡Regístrate!</button></a></li>
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-login\\" value=\\"/login\\">Iniciar Sesión</button></a></li>
    </ul>
  </div>
</header>

    <div id=\\"home-page\\">
      <div class=\\"banner\\">
        <p class=\\"banner-welcome\\">
          Cuidar de la naturaleza es cuidar de nosotros.
        </p>
        <button class=\\"btns-login btn-routing\\" id=\\"btn-register-movil\\" value=\\"/register\\">¡Regístrate!</button>
      </div>
      <div class=\\"process\\">
        <h2 id=\\"title-process\\">¿Cómo Funciona?</h2>
        <div class=\\"process-img\\">
          <img id=\\"process-movil\\" src=\\"img/process-movil.png\\" alt=\\"register\\">
          <img id=\\"process-desktop\\" src=\\"img/process-desktop.png\\" alt=\\"register\\">
        </div>
      </div>
    </div>
   
   </div><div>
  <header>
  <a href=\\"/\\"><img src=\\"img/logo-desktop.png\\" alt=\\"logotipo\\" id=\\"logoGF\\"></a>
  <span class=\\"menu-icon\\" id=\\"users-icon\\"><img id=\\"img-users\\" src=\\"img/users.png\\" alt=\\"\\"></span>
  <div class=\\"div-navegation\\">
    <ul id=\\"navegacion\\">
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-register\\" value=\\"/register\\">¡Regístrate!</button></a></li>
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-login\\" value=\\"/login\\">Iniciar Sesión</button></a></li>
    </ul>
  </div>
</header>
  <div id=\\"login-page\\">
     <form id=\\"login-form\\" class=\\"form-inicial\\">
       <h2 class=\\"titles\\" id=\\"title-form\\">¡Hola Green Friend!</h2>
       <img id=\\"heart-movil\\" src=\\"img/heart.png\\" alt=\\"heart\\">
       <label for=\\"email\\">Correo electrónico</label>
       <input type=\\"email\\" id=\\"login-email\\" placeholder=\\"alguien@ejemplo.com\\">
       <label for=\\"password\\">Contraseña</label>
       <div class=\\"show-passwords\\">
       <input type=\\"password\\" id=\\"login-password\\" placeholder=\\"******\\">
       <span class=\\"eyes-login\\">
       <i id=\\"show3\\" class=\\"fas fa-eye\\" style=\\"color: rgb(14, 99, 89); display: block;\\"></i>
       <i id=\\"hide3\\" class=\\"fas fa-eye-slash\\" style=\\"color: rgb(14, 99, 89); display: none;\\"></i>
       </span>
       </div>
       <br>
       <p id=\\"error-message\\"></p>
       <button id=\\"form-button-login\\" class=\\"submit-btn\\">Enviar</button>
       <button id=\\"btn-google\\" class=\\"submit-btn google\\">Entra con  <img src=\\"img/google.png\\" alt=\\"google\\" id=\\"google-icon\\"></button>
     </form>
     <div class=\\"img-register-desktop\\">
       <img id=\\"heart\\" src=\\"img/heart.png\\" alt=\\"heart\\">
     </div>
   </div>
  </div><div>
  <header>
  <a href=\\"/\\"><img src=\\"img/logo-desktop.png\\" alt=\\"logotipo\\" id=\\"logoGF\\"></a>
  <span class=\\"menu-icon\\" id=\\"users-icon\\"><img id=\\"img-users\\" src=\\"img/users.png\\" alt=\\"\\"></span>
  <div class=\\"div-navegation\\">
    <ul id=\\"navegacion\\">
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-register\\" value=\\"/register\\">¡Regístrate!</button></a></li>
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-login\\" value=\\"/login\\">Iniciar Sesión</button></a></li>
    </ul>
  </div>
 </header>
  <div id=\\"register-page\\">
     <form id=\\"register-form\\" class=\\"form-inicial\\">
      <img class=\\"leaf-img\\" src=\\"img/leafs-desktop.png\\" alt=\\"leafs\\">
       <h2 class=\\"titles\\" id=\\"title-form\\">Registro</h2>
       <label for=\\"email\\">Correo electrónico</label>
       <input type=\\"email\\" id=\\"user-email\\" placeholder=\\"alguien@ejemplo.com\\">
       <label for=\\"password\\">Contraseña</label>
       <div class=\\"show-passwords\\">
       <input type=\\"password\\" id=\\"user-password\\" placeholder=\\"Mínimo 6 carácteres\\">
       <span class=\\"eyes\\">
       <i id=\\"show1\\" class=\\"fas fa-eye\\" style=\\"color: #0e6359;\\"></i>
       <i id=\\"hide1\\" class=\\"fas fa-eye-slash\\" style=\\"color: #0e6359;\\"></i>
       </span>
       </div>
       <label for=\\"password\\"> Confirma Contraseña</label>
       <div class=\\"show-passwords\\">
       <input type=\\"password\\" id=\\"confirm-password\\" placeholder=\\"Mínimo 6 carácteres\\">
       <span class=\\"eyes-confirm\\">
       <i id=\\"show2\\" class=\\"fas fa-eye\\" style=\\"color: #0e6359;\\"></i>
       <i id=\\"hide2\\" class=\\"fas fa-eye-slash\\" style=\\"color: #0e6359;\\"></i>
       </span>
       </div>
       <br>
       <p id=\\"error-message-register\\"></p>
       <button id=\\"form-button-register\\" class=\\"submit-btn\\">Enviar</button>
       <button id=\\"btn-google\\" class=\\"submit-btn google\\"><img src=\\"img/google.png\\" alt=\\"google\\" id=\\"google-icon\\">Regístrate</button>
     </form>
     <div class=\\"img-register-desktop\\">
       <img id=\\"madre-tierra\\" src=\\"img/madreTierra.png\\" alt=\\"MadreTierra\\">
     </div>
   </div>
</div>"
`;

exports[`Coleccion de test sobre HOME should render 1`] = `
"<div>
  <header>
  <a href=\\"/\\"><img src=\\"img/logo-desktop.png\\" alt=\\"logotipo\\" id=\\"logoGF\\"></a>
  <span class=\\"menu-icon\\" id=\\"users-icon\\"><img id=\\"img-users\\" src=\\"img/users.png\\" alt=\\"\\"></span>

  <div class=\\"div-navegation\\">
    <ul id=\\"navegacion\\">
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-register\\" value=\\"/register\\">¡Regístrate!</button></a></li>
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-login\\" value=\\"/login\\">Iniciar Sesión</button></a></li>
    </ul>
  </div>
</header>

    <div id=\\"home-page\\">
      <div class=\\"banner\\">
        <p class=\\"banner-welcome\\">
          Cuidar de la naturaleza es cuidar de nosotros.
        </p>
        <button class=\\"btns-login btn-routing\\" id=\\"btn-register-movil\\" value=\\"/register\\">¡Regístrate!</button>
      </div>
      <div class=\\"process\\">
        <h2 id=\\"title-process\\">¿Cómo Funciona?</h2>
        <div class=\\"process-img\\">
          <img id=\\"process-movil\\" src=\\"img/process-movil.png\\" alt=\\"register\\">
          <img id=\\"process-desktop\\" src=\\"img/process-desktop.png\\" alt=\\"register\\">
        </div>
      </div>
    </div>
   
   </div>"
`;

exports[`Coleccion de test sobre LOGIN should render 1`] = `
"<div>
  <header>
  <a href=\\"/\\"><img src=\\"img/logo-desktop.png\\" alt=\\"logotipo\\" id=\\"logoGF\\"></a>
  <span class=\\"menu-icon\\" id=\\"users-icon\\"><img id=\\"img-users\\" src=\\"img/users.png\\" alt=\\"\\"></span>

  <div class=\\"div-navegation\\">
    <ul id=\\"navegacion\\" class=\\"show\\">
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-register\\" value=\\"/register\\">¡Regístrate!</button></a></li>
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-login\\" value=\\"/login\\">Iniciar Sesión</button></a></li>
    </ul>
  </div>
</header>

    <div id=\\"home-page\\">
      <div class=\\"banner\\">
        <p class=\\"banner-welcome\\">
          Cuidar de la naturaleza es cuidar de nosotros.
        </p>
        <button class=\\"btns-login btn-routing\\" id=\\"btn-register-movil\\" value=\\"/register\\">¡Regístrate!</button>
      </div>
      <div class=\\"process\\">
        <h2 id=\\"title-process\\">¿Cómo Funciona?</h2>
        <div class=\\"process-img\\">
          <img id=\\"process-movil\\" src=\\"img/process-movil.png\\" alt=\\"register\\">
          <img id=\\"process-desktop\\" src=\\"img/process-desktop.png\\" alt=\\"register\\">
        </div>
      </div>
    </div>
   
   </div><div>
  <header>
  <a href=\\"/\\"><img src=\\"img/logo-desktop.png\\" alt=\\"logotipo\\" id=\\"logoGF\\"></a>
  <span class=\\"menu-icon\\" id=\\"users-icon\\"><img id=\\"img-users\\" src=\\"img/users.png\\" alt=\\"\\"></span>
  <div class=\\"div-navegation\\">
    <ul id=\\"navegacion\\">
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-register\\" value=\\"/register\\">¡Regístrate!</button></a></li>
      <li><a><button class=\\"blue btn-routing\\" id=\\"btn-login\\" value=\\"/login\\">Iniciar Sesión</button></a></li>
    </ul>
  </div>
</header>
  <div id=\\"login-page\\">
     <form id=\\"login-form\\" class=\\"form-inicial\\">
       <h2 class=\\"titles\\" id=\\"title-form\\">¡Hola Green Friend!</h2>
       <img id=\\"heart-movil\\" src=\\"img/heart.png\\" alt=\\"heart\\">
       <label for=\\"email\\">Correo electrónico</label>
       <input type=\\"email\\" id=\\"login-email\\" placeholder=\\"alguien@ejemplo.com\\">
       <label for=\\"password\\">Contraseña</label>
       <div class=\\"show-passwords\\">
       <input type=\\"password\\" id=\\"login-password\\" placeholder=\\"******\\">
       <span class=\\"eyes-login\\">
       <i id=\\"show3\\" class=\\"fas fa-eye\\" style=\\"color: #0e6359;\\"></i>
       <i id=\\"hide3\\" class=\\"fas fa-eye-slash\\" style=\\"color: #0e6359;\\"></i>
       </span>
       </div>
       <br>
       <p id=\\"error-message\\"></p>
       <button id=\\"form-button-login\\" class=\\"submit-btn\\">Enviar</button>
       <button id=\\"btn-google\\" class=\\"submit-btn google\\">Entra con  <img src=\\"img/google.png\\" alt=\\"google\\" id=\\"google-icon\\"></button>
     </form>
     <div class=\\"img-register-desktop\\">
       <img id=\\"heart\\" src=\\"img/heart.png\\" alt=\\"heart\\">
     </div>
   </div>
  </div>"
`;
